### PR TITLE
caching synthesizer

### PR DESCRIPTION
**Pull request by dquang-tran**
https://linear.app/fleetworks/issue/FLE-71/add-synthesis-caching-for-vocode

### CODE REVIEW

File: vocode/streaming/synthesizer/caching_synthesizer.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        return self.inner_synthesizer.get_message_cutoff_from_voice_speed(message, seconds, words_per_minute)
+
+    async def create_speech(
+        self,
+        message: BaseMessage,
+        chunk_size: int,
+        bot_sentiment: Optional[BotSentiment] = None,
+    ) -> SynthesisResult:
+        result = self.redisClient.get(
+            cache_key(message.text, self.inner_synthesizer.get_synthesizer_config()))
```
Comment (theanht1) [inline vocode/streaming/synthesizer/caching_synthesizer.py:112]: **question**: Does `result` need to convert to `SynthesisResult`?

**Review by theanht1 — COMMENTED** [review]
In order to use GCP Redis for the Python service, we need to connect it to Cloud Run. You can check the reference here

- Docs: https://cloud.google.com/memorystore/docs/redis/connect-redis-instance-cloud-run
- Our serverless VPC connector: https://console.cloud.google.com/networking/connectors/details/us-central1/app-engine-redis-connecto?project=fleetworks

File: vocode/streaming/synthesizer/caching_synthesizer.py (hunk 1 above)
Comment (dquang-tran) [reply vocode/streaming/synthesizer/caching_synthesizer.py]: Good catch, pushed an old version.

**Comment by dquang-tran** [comment]
> In order to use GCP Redis for the Python service, we need to connect it to Cloud Run. You can check the reference here
> 
> * Docs: https://cloud.google.com/memorystore/docs/redis/connect-redis-instance-cloud-run
> * Our serverless VPC connector: https://console.cloud.google.com/networking/connectors/details/us-central1/app-engine-redis-connecto?project=fleetworks

Yep! That's done in the UI cloudbuild.

**Pull request merged**